apps/api/src/app/auth/usecases/password-reset-request/password-reset-request.usecase.ts

```diff
@@ -36,4 +46,58 @@ export class PasswordResetRequest {
       success: true,
     };
   }
+
+  private isRequestBlocked(user: UserEntity) {
+    const lastResetAttempt = user.resetTokenDate;
+
+    if (!lastResetAttempt) {
+      return {
+        isBlocked: false,
+        error: '',
+      };
+    }
+    const formattedDate = parseISO(lastResetAttempt);
+    const diffSeconds = differenceInSeconds(new Date(), formattedDate);
+    const diffHours = differenceInHours(new Date(), formattedDate);
+
+    if (diffHours < this.RATE_LIMIT_IN_HOURS && user?.resetTokenCount?.reqInDay >= this.MAX_ATTEMPTS_IN_A_DAY) {
```

Comment:
```suggestion
    const exceededHourRate = diffHours < this.RATE_LIMIT_IN_HOURS;
    const esceededDailyRate = user?.resetTokenCount?.reqInDay >= this.MAX_ATTEMPTS_IN_A_DAY;
    
    if (exceededHourRate && esceededDailyRate) {
```

Usually I prefer for those type of long compares to introduce a variable to help with readability, but up to you :)